cpp/include/cuspatial/types.hpp

```diff
@@ -16,46 +16,41 @@
 
 #pragma once
 
+#include <cstdint>
+
 namespace cuspatial {
 
 /**
- * @brief A 3D location: latitude, longitude, altitude
+ * @brief A 2D vector
  *
- * @tparam T the base type for the coordinates
- */
-template <typename T>
-struct location_3d {
-  T latitude;
-  T longitude;
-  T altitude;
-};
-
-/**
- * @brief A 2D Cartesian location (x, y)
+ * Used in cuspatial for both Longitude/Latitude (LonLat) coordinate pairs and Cartesian (X/Y)
+ * coordinate pairs. For LonLat pairs, the `x` member represents Longitude, and `y` represents
+ * Latitude.
  *
  * @tparam T the base type for the coordinates
  */
 template <typename T>
-struct coord_2d {
-  T x;
-  T y;
+struct alignas(2 * sizeof(T)) vec_2d {
```

Comment:
I think it would still be nice to enforce a type system on geometric type and geographic type - as algorithms only work on specific types. Proposing:

```c++
// Any coordinates
template <typename T>
struct alignas(2 * sizeof(T)) coord_2d {...}

// Longtitude/Latitude coordinates
template<typename T>
struct lonlat_2d : coord_2d<T> {}

// Cartesian coordinates
template<typename T>
struct cart_2d : coord_2d<T> {}
```

Reply:
cc @jrhemstad